src/Bootstrappers/BatchTenancyBootstrapper.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Bus\DatabaseBatchRepository;
+use Illuminate\Support\Facades\DB;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class BatchTenancyBootstrapper implements TenancyBootstrapper
+{
+    /**
+     * The database previous connection instance.
+     *
+     * @var \Illuminate\Database\Connection
+     */
+    protected $previousConnection = null;
+
+    public function bootstrap(Tenant $tenant)
+    {
+        $batchRepository = app(BatchRepository::class);
+
+        if ($batchRepository instanceof DatabaseBatchRepository) {
```

Comment:
Looks like `DatabaseBatchRepository` is the only type because the [interface binds to `DatabaseBatchRepository` concrete implementation in BusServiceProvider](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Bus/BusServiceProvider.php#L44). Also, Laravel batch achieved using database table only. 
<img width="681" alt="Screenshot 2022-09-13 at 11 49 55 AM" src="https://user-images.githubusercontent.com/54532330/189830594-9c89d1bf-dcdc-478d-950a-d518ed32a1d0.png">

Tests are passing after removing the `if` check. 
So the only reason to keep the check is maybe Laravel will add more options for Batching. Also, I found that `BatchRepositoryFake` also implements the same interface.

Reply:
> Also, I found that BatchRepositoryFake also implements the same interface.

Could you check if the reason for the `if` check was that? Meaning: if you try to fake the batch repository, and the if check is removed, then the code will run into some errors.

Reply:
> Could you check if the reason for the if check was that?

After looking into the source code, I see`BatchRepository ` is always resolved to `DatabaseBatchRepository` no matter if we use the Fake. I've tried `Bus::fake` in tests. I think that it makes sense because it looks like `BatchRepositoryFake` [is being used internally](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Support/Testing/Fakes/BusFake.php#L684-L689). 

Even after this, I'm still no 100% that we should remove the `if` check.

Reply:
Then try diving into that a bit deeper I guess, so that we're completely sure what code we should use here.

Reply:
So I dug it up again today. In class, the interface always resolves to `DatabaseBatchRepository` no matter whether I do `Bus::fake` or not. When faking Bus, use the `BatchRepositoryFake` [internally](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Support/Testing/Fakes/BusFake.php#L684-L689). 

Since, I'm not able to be 100 clear that if interface would resolve anything else, I talked to Lukas. 

<img width="756" alt="image" src="https://user-images.githubusercontent.com/54532330/191261295-e1288539-9d32-4ba6-bd1f-9b057e465642.png">

So we might want to keep the check because of the connection getter and setter. Also, I liked his suggestion:
> So either I'd keep the if, or make the $batchRepository property DatabaseBatchRepository

Reply:
@lukinovec 

> In that case, I'd definitely keep the if.

If it's only present on class X, **and the variable can also only ever be type X**, then the `if` is unnecessary. But I guess it can help with IDE support and narrowing the type within that scope.

It's fine if there are more implementations of the interface, but I'm wondering if there are any cases when this would resolve as those other types.

Regarding the idea to hardcode the `DatabaseBatchRepository` type, that *could* work but there are more things to keep in mind:
1. It's possible that **the correct way to resolve the batch repository is to resolve the interface**. For example, the interface may be bound as a singleton, or have `->extend` bindings etc. With `DatabaseBatchRepository` only being the class which implements the interface, but **not the class that has this "important position" in the service container**.
2. I generally don't like typehinting implementations in case someone used a different one (e.g. wrote a `RedisBatchRepository`), but the `if` check has the same problem, so typehinting isn't any worse. Also, people can just extend or replace the `BatchTenancyBootstrapper`

Regarding point 1), I would:
1. Try changing the typehint and seeing if the code still works
2. Dive into Laravel source code to see if the interface has any special status in the service container, compared to the DB implementation class

Reply:
I think we should not typehint `DatabaseBatchRepository ` mostly because of Samuel's "2. I generally don't like..." point. 
The main issue here is whether keep the `if` check or not. It depends on the interface resolution. As I stated above, according to my discovery, I found the interface bound singleton to only the `DatabaseBatchRepository` class. The 
The `BatchRepositoryFake` class also implements the interface, but it never resolved this class. @lukinovec can you please try what other classes the interface can resolve and make sure if we can remove the `if` check? You can read my replies above for more information.

Reply:
Seems like the interface has no special logic (e.g. `->extend` calls), it simply resolves to the DB repository:

```php
protected function registerBatchServices()
{
    $this->app->singleton(BatchRepository::class, DatabaseBatchRepository::class);

    $this->app->singleton(DatabaseBatchRepository::class, function ($app) {
        return new DatabaseBatchRepository(
            $app->make(BatchFactory::class),
            $app->make('db')->connection($app->config->get('queue.batching.database')),
            $app->config->get('queue.batching.table', 'job_batches')
        );
    });
}
```

So my view is that:
1. In theory, typehinting the interface would be more correct, and then the `if ()` check adds more IDE support and help for static analysis
2. In practice, typehinting the interface and pairing it with an `if ()` check is a bit messy. When it comes to the behavior, typehinting the DB implementation will probably work because they're *both* singletons

So ultimately: typehinting the DB implementation is a bit less correct, but it's more clear. Therefore: **assuming everything works well with that approach**, we'll do that.

Try changing the typehint to use `DatabaseBatchRepository`, and if everything works well, commit it here.

Reply:
> Try changing the typehint to use DatabaseBatchRepository, and if everything works well, commit it here.

Just committed the changes. Everything is working, and the tests are passing.